Plugin did not respond error returned when calling region data provider

### Terraform Version
Terraform v1.3.2

### Affected Resource(s)
- region_data

### Expected Behavior
region_data return expected result.

### Actual Behavior
```
kunliu@LAPTOP678-MA-US region_data % terraform plan
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - hashicorp/biganimal in /Users/kunliu/.terraform.d/plugins/hashicorp.com/edu/biganimal/0.3.1/darwin_arm64
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
var.cloud_provider
  Cloud Provider

  Enter a value: aws

data.biganimal_region.this: Reading...
╷
│ Error: Plugin did not respond
│ 
│   with data.biganimal_region.this,
│   on data.tf line 10, in data "biganimal_region" "this":
│   10: data "biganimal_region" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-biganimal plugin:

panic: Invalid address to set: []string{"regions", "0", "id"}

goroutine 40 [running]:
github.com/EnterpriseDB/terraform-provider-biganimal/pkg/utils.SetOrPanic(0x1400021a500?, {0x104ef31e5, 0x7}, {0x105067a60?, 0x1400019c630?})
	/Users/kunliu/repos/terraform-provider-biganimal/pkg/utils/props.go:102 +0x340
github.com/EnterpriseDB/terraform-provider-biganimal/pkg/provider.(*RegionData).Read(0x105188b80?, {0x105188b48, 0x14000130b40}, 0x1176592e000?, {0x1050b2fc0?, 0x140004f5ec0?})
	/Users/kunliu/repos/terraform-provider-biganimal/pkg/provider/data_source_region.go:103 +0x2a0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14000434700, {0x105188b80, 0x1400019b140}, 0xd?, {0x1050b2fc0, 0x140004f5ec0})
	/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:724 +0xec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0x14000434700, {0x105188b80, 0x1400019b140}, 0x140000c0880, {0x1050b2fc0, 0x140004f5ec0})
	/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:943 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0x1400000f1b8, {0x105188b80?, 0x1400019b020?}, 0x1400007e120)
	/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1179 +0x304
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0x140000d06e0, {0x105188b80?, 0x1400019a870?}, 0x1400004e140)
	/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658 +0x36c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x105153ee0?, 0x140000d06e0}, {0x105188b80, 0x1400019a870}, 0x140002fc000, 0x0)
	/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400044e000, {0x10518b6a0, 0x14000472000}, 0x14000156120, 0x14000448a80, 0x1055fcff0, 0x0)
	/Users/kunliu/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1295 +0x9d8
google.golang.org/grpc.(*Server).handleStream(0x1400044e000, {0x10518b6a0, 0x14000472000}, 0x14000156120, 0x0)
	/Users/kunliu/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1636 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/kunliu/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:932 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/kunliu/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:930 +0x298

Error: The terraform-provider-biganimal plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/examples/region_data/data.tf
+++ b/examples/region_data/data.tf
@@ -9,7 +9,8 @@ terraform {
 
 data "biganimal_region" "this" {
   cloud_provider = var.cloud_provider
-  region_id = "us-east-1"
+  region_id = var.region_id
+
 }
 
 output "regions" {

--- a/examples/region_data/variables.tf
+++ b/examples/region_data/variables.tf
@@ -7,3 +7,8 @@ variable "cloud_provider" {
     error_message = "Please select one of the supported regions: aws, azure."
   }
 }
+
+variable "region_id" {
+  type        = string
+  description = "region id"
+}

--- a/pkg/models/region.go
+++ b/pkg/models/region.go
@@ -1,7 +1,7 @@
 package models
 
 type Region struct {
-	Id        string `json:"regionId,omitempty" mapstructure:"id"`
+	Id        string `json:"regionId,omitempty" mapstructure:"region_id"`
 	Name      string `json:"regionName,omitempty" mapstructure:"name,omitempty"`
 	Status    string `json:"status,omitempty" mapstructure:"status,omitempty"`
 	Continent string `json:"continent,omitempty" mapstructure:"continent,omitempty"`

